### ISSUE TITLE
Terminal file link now opens file search instead of file itself

### ISSUE THREAD

**Issue by ak99372** [microsoft/vscode#106520]
- VSCode Version: 1.48.2
- OS Version: Win 10.0.18363

Steps to Reproduce:

1. When there is a reference/path to a specific file in terminal (e.g. compilation error)

![image](https://user-images.githubusercontent.com/9924769/92977518-18d73f80-f442-11ea-9e19-4d438626843c.png)

2. [Ctrl] + [click] on the file reference opens **file search** instead of the file directly.

![image](https://user-images.githubusercontent.com/9924769/92977668-6eabe780-f442-11ea-88db-46179d8fd279.png)

**Issue/Expected behavior**: When opening file of a specific path there is always going to be just 1 such file so it should open directly from terminal. This is how it used to work few releases back (no reason to click on two different places just to open a single file). 

Alternatively is there a setting that can set this behavior?

Does this issue occur when all extensions are disabled?: Yes

**Comment by Tyriar** [comment]
The "validated" links only happen when the path is relative to the opened folder, it would work in your example if it was prefixed with `AzureFunctions/`. No plans to change this.

> This is how it used to work few releases back (no reason to click on two different places just to open a single file).

A few releases back it wouldn't offer a link at all (unless fully qualified).

**Comment by ak99372** [comment]
@Tyriar thanks for prompt reply. I did a bit of digging and you're right this is related to the folder `AzureFunctions`.

- When I right click in the Side Bar to [Open In integrated Terminal] directly on this folder -> all validated links do **open the file directly**.

- However when I open terminal in a different folder and then navigate to this folder (via `cd AzureFunctions`) then the search is opened instead.

In other words Terminal seems to keep some sort of reference to the folder it was **originally opened** from, instead of changing when the current folder changes. Not sure what the underlying reason to keep such references to "open folder" would be but since the original path isn't something shown (or changeable) this creates inconsistent behavior.

**Comment by Tyriar** [comment]
Currently we don't plan on changing this, for Windows specifically we've been told by the Windows kernel team that it's not possible for us to determine the working directory of a terminal session. For Linux/macOS it's possible (and we do it to drive `"terminal.integrated.splitCwd": "inherited"`), but we don't want to introduce differing behavior across platforms.
